orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/pipeline/MonitorPipelineStage.java

```diff
@@ -0,0 +1,135 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.orca.front50.pipeline;
+
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.netflix.spinnaker.kork.artifacts.model.ExpectedArtifact;
+import com.netflix.spinnaker.orca.api.pipeline.graph.StageDefinitionBuilder;
+import com.netflix.spinnaker.orca.api.pipeline.graph.TaskNode;
+import com.netflix.spinnaker.orca.api.pipeline.models.ExecutionStatus;
+import com.netflix.spinnaker.orca.api.pipeline.models.StageExecution;
+import com.netflix.spinnaker.orca.front50.tasks.MonitorPipelineTask;
+import com.netflix.spinnaker.orca.pipeline.persistence.ExecutionRepository;
+import com.netflix.spinnaker.orca.pipeline.tasks.artifacts.BindProducedArtifactsTask;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nonnull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MonitorPipelineStage implements StageDefinitionBuilder {
+
+  private final Logger log = LoggerFactory.getLogger(getClass());
+
+  public static final String PIPELINE_CONFIG_TYPE =
+      StageDefinitionBuilder.getType(MonitorPipelineStage.class);
+
+  final ExecutionRepository executionRepository;
+
+  @Autowired
+  public MonitorPipelineStage(ExecutionRepository executionRepository) {
+    this.executionRepository = executionRepository;
+  }
+
+  @Override
+  public void taskGraph(@Nonnull StageExecution stage, @Nonnull TaskNode.Builder builder) {
+    StageParameters stageData = stage.mapTo(StageParameters.class);
+
+    builder.withTask("monitorPipeline", MonitorPipelineTask.class);
+
+    if (stageData.expectedArtifacts != null) {
+      builder.withTask(BindProducedArtifactsTask.TASK_NAME, BindProducedArtifactsTask.class);
+    }
+  }
+
+  public enum MonitorBehavior {
+    /**
+     * During monitor, wait for all pipelines to complete before completing the stage even if a
+     * monitored pipeline fails
+     */
+    WaitForAllToComplete,
+
+    /** As soon as (at least) one monitored pipeline fails, terminate this stage */
+    FailFast
+  }
+
+  public static class StageParameters {
+    /**
+     * DO NOT USE: Legacy from PipelineStage Use {@link executionIds} instead (even if you only have
+     * one execution ID to monitor)
+     */
+    public String executionId;
```

Comment:
I'm confused. How would anyone know not to use this unless they're sleuthing code? At minimum, I'd say we need `@Deprecated` & `@DeprecationInfo` on this. To be tackled later, I'd like a means of being able to provide this feedback of deprecation to end users (perhaps via `PipelineExecution.systemNotifications`?)

Reply:
thanks, updating